### PR TITLE
ensure kwargs are always in sync across the whole application

### LINKED ISSUE #7786: [REGRESSION again] Accordion close behavior back in gradio 4.22.0
Accordion either pops open and while streaming cannot be closed, or other way it closes and cannot be opened, depending upon where the elements are.

Reproduction
pops-open no matter what repro is below. If you need a repro of case where it pops-close no matter, let me know.

```python
import time

import gradio as gr

with gr.Blocks() as demo:
    with gr.Column():
        with gr.Accordion():
            image = gr.Image()
        with gr.Tabs():
            with gr.Tab("Chat"):
                chatbot = gr.Chatbot()
                msg = gr.Textbox()
                clear = gr.Button("Clear")
            with gr.Tab("Other"):
                msg2 = gr.Textbox()

    def user(user_message, history):
        return "", history + [[user_message, None]]

    def bot(history):
        history[-1][1] = ''
        for msg in range(0, 10000):
            history[-1][1] += str(msg)
            if msg % 100 == 0:
                history[-1][1] += '\n'
            time.sleep(0.01)
            yield history

    msg.submit(user, [msg, chatbot], [msg, chatbot], queue=False).then(
        fn=bot, inputs=chatbot, outputs=chatbot, api_name='bot',
    )
    clear.click(lambda: None, None, chatbot, api_name='clear')

demo.queue()
demo.launch()
```

### LINKED ISSUE #7895: regression on 4.22.0 and following: .focus() doesn't work anymore inside an Accordion(open=False)
### Describe the bug

Thank you for the tremendous work on Gradio!

I use Accordion(open=False) to hide components (such as a Dropdown and some buttons) and show them only when the user need it.
I then use a .focus event on the Dropdown.
It worked perfectly with gradio 4.21.0 but since 4.22.0, 4.23.0 and 4.24.0, it doesn't work anymore : it is impossible to open the Dropdown when I listen to gr.Dropdown.focus() inside my Accordion (only when this Accordion is initially set to open=False), since the Accordion closes itself automatically

this prevents me from using 4.22.0 and newer versions

and another idea : it would be nice to have an open=True parameter for gr.Dropdown as in Accordions

Thanks a lot!

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

    import gradio as gr
    
    with gr.Blocks() as demo:
    
        with gr.Accordion(open=False):
            zouzou = gr.Dropdown(
                choices=["zouzou", "zouzou2"],
                type="index",
            )
    
        zouzou.focus(
            fn=lambda x: print(x),
            inputs=[zouzou],
            outputs=[],
            queue=False,
        )
    
    demo.launch()

### Screenshot

before I click on the Dropdown
image

and then the Accordion closes itself when I click on the Dropdown, which disappears
image

### Logs

```shell
No logs
```

### System Info

```shell
gradio==4.22.0 (or 4.23.0 or 4.24.0)
Mac Safari or Chrome 
PC Chrome or Edge
iPhone Safari or Chrome
```

### Severity

I can work around it

### PR DESCRIPTION
**Pull request by pngwn**
## Description

Closes #7786. Closes #7895.

The issue was not present in dev mode but _was_ present in the built files. So we weren't noticing this behaviour day to day in development.

The root cause, seems to be that the list of props that we were using to listen for changes wasn't the same in dev + build. I'm not 100% why really, its a little confusing but I have confirmed that the approach in this PR works for all modes.

I still need to look at the test, but I need to create a failing test from main first. Will update shortly but the code is good for review.

To test you can use the `blocks_flipper` demo. This does not behave itself on main but works correctly in dev. You can also use the repro in #7895 as it is the same issue.

Be sure to build the frontend locally and visit port `7860` and not the dev port (9876). 

**Update**

I've update the test. The test was opening the accordion, clicking the slider and then checking that accordion text was on the page. The issue here is that for a brief moment after changing the slider value this is true, so sometimes this was passing (sometime it was failing too but we have retries on).

I've switched it out for a test that opens the accordion, changes the slider value then enters text into the textbox and waits for the server response + state update. This guarantees that the UI state has fully settled before asserting on the accordion contents.

This test passes on this branch but it fails consistently on main.

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Can confirm this fixes the bug @pngwn ! Can this demo be added to 2e2 tests?

File: js/app/src/RenderComponent.svelte [hunk 1]
```diff
@@ -28,9 +28,7 @@
 			construct(_target, args: Record<string, any>[]) {
 				//@ts-ignore
 				const instance = new _target(...args);
-				const props = Object.getOwnPropertyNames(instance).filter(
-					(s) => !s.startsWith("$")
```
[the remark this answers is not in the document]
Comment (pngwn) [reply js/app/src/RenderComponent.svelte:32]: In addition to the prop names, there are also some internal svelte things (methods mostly) that start with `$`. Was always a little hacky.

**Comment by pngwn** [comment]
@freddyaboulton Which demo? The blocks flipper is the original test case hear and is in the e2e tests, or do you mean the streaming one?

**Comment by freddyaboulton** [comment]
Oh weird I didn't know we already had a test! I guess `toBeVisible` is not the right playwright assertion to use? The test should not be passing right?

**Comment by freddyaboulton** [comment]
Good to merge, we can take a look at the test later.

**Comment by pngwn** [comment]
I updated that test so that it doesn't / would fail consistently on main. The assertion was okay, its just that the test was running faster than the UI could update. This is why it was flakey. Funnily enough the flake was the 'correct' behaviour.

> I've updated the test. The test was opening the accordion, clicking the slider and then checking that accordion text was on the page. The issue here is that for a brief moment after changing the slider value, this is true. So sometimes this was passing (sometimes it was failing too but we have retries on).
>
> I've switched it out for a test that opens the accordion, changes the slider value then enters text into the textbox and waits for the server response + state update. This guarantees that the UI state has fully settled before asserting on the accordion contents.
>
> This test passes on this branch but it fails consistently on main.

**Comment by freddyaboulton** [comment]
Nice thanks @pngwn !

**Pull request merged**